docs/BR.md

```diff
@@ -1756,142 +1754,916 @@ The CA SHALL enforce multi-factor authentication for all accounts capable of dir
+| __Field__                  | __Presence__ | __Description__ |
+| ---                        | -            | ------          |
+| `certificatePolicies`      |              |                 |
+| \ \ **1**                  | MUST         | The first `PolicyInformation` present in the `certificatePolicies`. |
+| \ \ \ \ `policyIdentifier` |              | A Reserved Certificate Policy Identifier (see [Section 7.1.6.1](#7161-reserved-certificate-policy-identifiers)). |
+| \ \ \ \ `policyQualifiers` | SHOULD NOT   | See below for restrictions on `policyQualifiers` |
+| \ \ **2+**                 | SHOULD       | The CA SHOULD include additional Reserved Certificate Policy Identifiers for each type of Subscriber certificate that may be issued beneath it. |
+| \ \ \ \ `policyIdentifier` | MUST         | A Reserved Certificate Policy Identifier (see [Section 7.1.6.1](#7161-reserved-certificate-policy-identifiers)). |
+| \ \ \ \ `policyQualifiers` | SHOULD NOT   | See below for restrictions on `policyQualifiers` |
+| \ \ **3+**                 | MAY          | The CA MAY include additional `PolicyInformation` values that meet the following profile: |
+| \ \ \ \ `policyIdentifier` | MUST         | An identifier documented by the CA in its Certificate Policy and/or Certification Practice Statement. |
+| \ \ \ \ `policyQualifiers` | SHOULD NOT   | See below for restrictions on `policyQualifiers` |
+| \ \ **4**                  | MUST NOT     | The CA MUST NOT include any additional `PolicyInformation` values that do not meet the above profile. |
 
-   For all other Subordinate CA Certificates, including Technically Constrained Subordinate CA Certificates:
+If the `policyQualifiers` is permitted and present within a `PolicyInformation` field, it MUST be formatted as follows:
 
-   This extension MUST be present and SHOULD NOT be marked critical[^**].
+| __Qualifier ID__                     | __Presence__ | __Field Type__ |  __Contents__ |
+| ---                                  | -            | -              | -----         |
+| `id-qt-cps` (OID: 1.3.6.1.5.5.7.2.1) | MAY          | `IA5String`    | The HTTP or HTTPS URL for the Issuing CA's Certificate Policies, Certification Practice Statement, Relying Party Agreement, or other pointer to online policy information provided by the Issuing CA. |
+| Any other qualifier                  | MUST NOT     | -              | -             |
+
+##### 7.1.2.8.6 Extended Key Usage - Non-TLS CA
+
+The Issuing CA MUST verify that the Subordinate CA Certificate is authorized to issue certificates for each included extended key usage purpose. Multiple, independent key purposes (e.g. `id-kp-timeStamping` and `id-kp-codeSigning`) SHOULD NOT be present.
```

Comment:
Regarding the statement "...for each included extended key usage purpose."

In general, CA may issue leaf certs for EKUs not in the CA certificate, especially in the "Non-TLS CA" case, so what's the value of this MUST requirement?  While the issuing CA of a subordinate CA could verify the the inclusion of what goes into the subordinate CA certificate, that doesn't really help if the subordinate CA can issue certs with more EKUs than the Issuing CA authorized.  

If the Issuing CA wants to permit the Subordinate CA to issue TLS, S/MIME or Code Signing, then yes, those should/must be specifically authorized.  Does that need to be in the BRs, or is that a root program requirement for those that permit these uses?  We should delete the MUST or limit to those EKUs that are "important" to control.

Reply:
> In general, CA may issue leaf certs for EKUs not in the CA certificate,

No they can’t. Maybe I’m misunderstanding you, but this gets rejected by a variety of software, because of EKU chaining, most notably, Microsoft’s.

> that doesn't really help if the subordinate CA can issue certs with more EKUs than the Issuing CA authorized

right, which is why well-behaving software like Microsoft CAPI or Mozilla NSS or Chrome reject this. OpenSSL does as well, if you extend with a custom trustpurpose for the EKU.

I don’t understand the rest of your question, because this is an existing requirement in the BRs today. However, perhaps I’m misunderstanding you?

Reply:
@CBonnell Perhaps you could explain here why you feel this discussion is at all relevant to this particular section? It does not appear that the requirement you're arguing against exists, but I'm hoping you can be precise here. As I mentioned, the requirement *in this section* that you're commenting on is an existing requirement (as mentioned in the overall summary here)

Reply:
My interpretation of your comment "No they can’t. Maybe I’m misunderstanding you, but this gets rejected by a variety of software, because of EKU chaining, most notably, Microsoft’s." is that you were asserting it's mis-issuance if a CA (name + key pair) issues a certificate with an EKU value that does not appear in any certificates corresponding with that CA. I think that's what Doug was alluding to as well, but perhaps I misunderstood.

Reply:
There are plenty of enterprise cases where CAs add things like the MicrosoftTemplate EKU but that is not in the issuing CA list of EKUs and the TLS certs are trusted by all TLS clients that I'm aware of - even MS.  If we want to include some root program requirements in the BRs and list the EKUs that are processed by some client applications in a special way and we want to apply the requirement above when those are included , then let's discuss.

Reply:
@CBonnell Thanks. I was not saying it’s misissuance, and I was pointing out that these requirements don’t currently prohibit/require it.

@dougbeattie To make sure we’re on the same page:
- This section says that, for a sub-CA, the issuing CA needs to verify the EKUs it includes, whatever those EKUs are. Are you saying you object to “verify what is actively included in the sub-CA”?
- Separately, you’re discussing in the comments here the EKUs for end-entity certificates. End-entity certificates extensions are dealt with in 7.1.2.6, with all types referring to the extension profile in 7.1.2.6.6, which refers to 7.1.2.6.10 for the EKU for EE certs. That’s why I’m having trouble why you’re raising an issue about EE certs on a subCA section, and hoping you can expand more.

Related, I’m trying to understand if you simply got confused, to better see if there are ways for document layout so that you aren’t confused. I had thought this was clear, but I could be both misunderstanding the concern or the source of confusion here.

Reply:
> This section says that, for a sub-CA, the issuing CA needs to verify the EKUs it includes, whatever those EKUs are. Are you saying you object to “verify what is actively included in the sub-CA”?

Of course the Issuing CA must validate and understand all the fields in the Subordinate CA certificate so calling out a single extension as a new MUST seems out of place, especially given that a) the CA itself never uses those EKUs, and b) that it does not restrict what EKUs can be in the leaf certs.

>Separately, you’re discussing in the comments here the EKUs for end-entity certificates. End-entity certificates extensions are dealt with in 7.1.2.6, with all types referring to the extension profile in 7.1.2.6.6, which refers to 7.1.2.6.10 for the EKU for EE certs. That’s why I’m having trouble why you’re raising an issue about EE certs on a subCA section, and hoping you can expand more.

Yes, the only reason to include EKUs in the Subordinate CA is to limit what can go into the EE certs.  The specific section I'm commenting on is for non-TLS, so I'm not sure why you're referencing subscriber EE profiles as those are not the ones that this type of CA would need to comply with.  What's the value of this requirement and what are we looking to prevent?

> Related, I’m trying to understand if you simply got confused, to better see if there are ways for document layout so that you aren’t confused. I had thought this was clear, but I could be both misunderstanding the concern or the source of confusion here.

I'm not sure why we're adding a MUST (something that will need to be specifically logged and audited) for verifying the list of EKUs in a Subordinate CA when it does not restrict what that CA can do or issue (again, except for those client applications that do application specific/non standard EKU "chaining" for a small subset of EKUs)

Reply:
Hey Doug, maybe it'd be easier to hop on a phone call to hash out? Your latest reply still seems confused, and I'd love to try and help sort that.

For example, you say
> so calling out a single extension as a new MUST seems out of place

This isn't calling out a single extension as a MUST. Literally the entire work of the profiles is saying this for each and every field, and setting expectations here about the level of validation expected for each and every field. I'm not sure what you feel is unique to the extension here, and I hope I'm just misunderstanding, because it seems wildly unfounded?

> What's the value of this requirement and what are we looking to prevent?

I've repeatedly stated to you: this is an existing requirement in the BRs. This is not something new from profiles, this is the logical interaction of our existing requirements.

I'm not sure a point-by-point takedown of your other bits would be that useful to helping us reach understanding. I'm concerned, for example, of your interpretation of MUST having impact on logging and auditing, as the logging and auditing requirements would apply to *all* extensions and *all fields* that allow CA-provided values. But I figure if we have a chance to discuss, this might be easier, because I think there's just some confusion about this requirement, and I want to both address it, and make sure we've got wording to prevent similar confusion in other sections.

Reply:
OK, Doug and I had a chance to sync up offline, and here's my notes of the concerns:
- In 7.1.2.8.7 (TLS CA), we have no rules on the "Any other value" (or on `id-kp-clientAuth`, but that's less important), while in 7.1.2.8.6 (non-TLS CA), we're explicitly setting rules on "Any other value" *and* `codeSigning`, `emailProtection`, `timeStamping`.
  - Part of this I mentioned is to be covered by the (existing/old BRs) 7.1.2.4 catch-all, which is currently titled "7.1.2.4 TO DELETE" in the current version (reflecting the fact that I need to reword this section). That is, for _any_ extension or field where we allow (MAY, SHOULD NOT) "any other value", the idea is that should refer to a common section that expresses our (existing) rules
  - However, our existing rules don't require the CA to "verify the Subordinate CA Certificate is authorized to issue" - merely, that the issuing CA "is aware of a reason for including that data in the certificate" and that will not "mislead a Relying Party about the certificate information verified by the CA".
  - That said, Existing/Old 7.1.2.4 repeatedly discusses `extKeyUsage`, and with a requirement that either the EKU "falls within an OID arc for which the Applicant demonstrates ownership" (7.1.2.4 (a)(i)) or "the Applicant can otherwise demonstrate the right to assert the data in a public context" (7.1.2.4 (a)(ii).
  - So one area of question: Is the new expression ("authorized to issue certificates for each included extended key usage purpose") more, less, or the same as the current expression (within 7.1.2.4)
- Separately, we have Existing/Old 7.1.2.2 (g). This _explicitly_ places requirements on Cross Certificates that the issuing CA must verify (nearly word for word identical with the language in this section), but does not have identical language for this non-TLS _non-cross-certificate_ sub-CA
  - If we remove the language here (in new 7.1.2.8.6) regarding validation, this creates a bit of a sequencing issue, namely:
    - If I create root, then non-TLS cross-certificate, it's unambiguous in the existing BRs that the issuing CA has to validate the EKUs on the cross-certificate.
    - If I create a non-TLS sub-CA first, can I include any EKU that I'm "aware of a reason for including" without further validation (that is, ignoring old 7.1.2.4 (a)), and then transfer that to a third-party, who generates a root? In that sequence of events, we go from a state of (non-TLS sub-CA) to (root, non-TLS cross-certified sub-CA) - that is, the creation of "root" makes the sub-CA become a cross-certificate - except the non-TLS cross-certificate wouldn't have followed the rules for cross certificates.

In terms of next steps:
- I have a concrete action item already on my TODO list, which is every place we have "Any other value", to refer to a section that is functionally equivalent to old-7.1.2.4 (this is called "7.1.2.4 TO DELETE" in the current draft, because it needs to be reworded)

What we need to determine is:
1. Do we agree that old-7.1.2.4 (a)'s requirements are functionally equivalent to new-7.1.2.8.6's explicit language, or do we believe 7.1.2.8.6 is expressing something more restrictive than the logical consequence of old-7.1.2.4 (a)?
2. Do we remove the list of explicitly permitted EKUs that are MAY in 7.1.2.8.6 (`clientAuth`, `codeSigning`, `emailProtection`, `timeStamping`), since they're the same as "Any other value"
3. Do we need specific language for EKUs in this section (to reiterate/clarify old/new-7.1.2.4), or should "Any other value" simply refer to new-7.1.2.4 (whatever form it takes)?
4. If we don't fold `clientAuth`-et-al into "any other value", what rules apply - Do the new-7.1.2.4 rules apply, or are these four values special and unique?

Reply:
> I have a concrete action item already on my TODO list, which is every place we have "Any other value", to refer to a section that is functionally equivalent to old-7.1.2.4 (this is called "7.1.2.4 TO DELETE" in the current draft, because it needs to be reworded)

Agree, we need a common section for this.  

>What we need to determine is:

>Do we agree that old-7.1.2.4 (a)'s requirements are functionally equivalent to new-7.1.2.8.6's explicit language, or do we believe 7.1.2.8.6 is expressing something more restrictive than the logical consequence of old-7.1.2.4 (a)? That is, is "right to assert" the same as "authorized to issue certificates" (for EKUs), or are these different concepts?

If we add a MUST to the new language then it's not quite equivalent.  We should make the statement identical or as close as possible to the current BRs for this release and if we find a need to tighten them up in V2, we can.

>Do we remove the list of explicitly permitted EKUs that are MAY in 7.1.2.8.6 (clientAuth, codeSigning, emailProtection, timeStamping), since they're the same as "Any other value"

We have the same dilemma in the OCSP signing cert where we list a set of EKUs with MUST NOT, and then say MUST NOT for "any others".  My feeling is that if we have a specific MUST/MUST NOT statement for "any other", then we don't need to list specific values with the same statement.  My preference is to remove "redundant" items in the lists.

>Do we need specific language for EKUs in this section (to reiterate/clarify old/new-7.1.2.4), or should "Any other value" simply refer to new-7.1.2.4 (whatever form it takes)?

I recommend we use the same statement/reference for all "any other" objects. 

> If we don't fold clientAuth-et-al into "any other value", what rules apply - Do the new-7.1.2.4 rules apply, or are these four values special and unique?

They are special, but is it worth calling them out without saying something special about them?  As far as the BRs are concerned, they are just another non serverAuth EKU and can be treated the same way, even though some root programs place specific requirements on some of them.  I recommend removing them from this section.

As far as the statement about cross certificates where the" Existing/Old 7.1.2.2 (g) explicitly places requirements on Cross Certificates that the issuing CA must verify", I'd leave that as is.  Roots signing cross certificates needs to be done with care and very close attention to detail because this will enable or prevent that cross signed root from issuing certain types of certificates (per Application Software Suppliers implementations) .

Reply:
> If we add a MUST to the new language then it's not quite equivalent. 

I very much disagree here, and so I'm curious to understand how you see this as not an existing requirement. 7.1.2.4(a) is a SHALL, SHALL and MUST are the same, and this is simply stating the existing expectation. What do you see as the difference between how the existing 7.1.2.4(a) works? I guess this means, in the context of an EKU, which 7.1.2.4(a) is unambiguous about, how is "right to assert" an EKU not 100% equivalent to "authorized to issue"?

Reply:
OK, will agree.